src/main/java/internet/shop/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package internet.shop.service.impl;
+
+import internet.shop.dao.CarDao;
+import internet.shop.lib.Inject;
+import internet.shop.lib.Service;
+import internet.shop.model.Car;
+import internet.shop.model.Driver;
+import internet.shop.service.CarService;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).orElseThrow(()
+                -> new RuntimeException("Not exist id"));
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(Car car) {
+        if (carDao.update(car) == null) {
+            throw new RuntimeException("Not exist car");
+        }
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        car.setDrivers(driver);
+        if (!carDao.get(car.getId()).isEmpty()) {
+            carDao.update(car);
+            return;
+        }
+        carDao.create(car);
```

Comment:
the same with removeDriverFromCar()
```suggestion
        car.getDrivers().add(driver);
        carDao.update(car);
```

Reply:
done